src/pages/index.js

```diff
@@ -34,75 +44,49 @@ const enableValidation = (settings) => {
 };
 
 function handleImageFormSubmit(name, link) {
-  addImagePopup.renderLoading(true);
-  api
-    .addNewCard({ name, link })
-    .then((newCardData) => {
-      const { name, link } = newCardData;
-      cardSection.addItem(
-        new Card(
-          newCardData,
-          "#card-template",
-          imagePopup,
-          (title, link) => imagePopup.open(title, link),
-          (cardElement, cardID) => {
-            deleteImagePopup.open(cardElement, newCardData._id);
-          },
-          api
-        ).getView()
-      );
-    })
-    .catch((err) => console.error("Error adding new card:", err));
-  addImagePopup.close();
+  function makeRequest() {
+    return api.addNewCard({ name, link }).then(renderCard);
+  }
+  handleSubmit(makeRequest, addImagePopup);
 }
 
-function handleProfileFormSubmit(name, description) {
-  editProfilePopup.renderLoading(true);
-  api
-    .updateUserInfo({ name: name, about: description })
-    .then((updatedUser) => {
+function handleProfileFormSubmit(name, about) {
+  function makeRequest() {
+    return api.updateUserInfo({ name, about }).then((updatedUser) => {
       userInfo.setUserInfo({
         name: updatedUser.name,
-        job: updatedUser.about,
+        about: updatedUser.about,
+        avatar: userInfo.getUserInfo().avatar,
```

Comment:
`        avatar: userInfo.getUserInfo().avatar,
`

You have everything in `updatedUser` (including the avatar)


```js

      userInfo.setUserInfo(updatedUser);

```

Reply:
I've fixed this part. I did run into a similar issue in my handleAvatarSubmit, but don't call updatedUser there since it returns the avatar link directly. Should I use userInfo.getUserInfo() in this case?

Reply:
> Should I use userInfo.getUserInfo() in this case?

`userInfo.getUserInfo()` is not needed here because you get the whole object from the server with all 4 params (including avatar)